dipy/tracking/local/localtracking.py

```diff
@@ -93,12 +100,22 @@ def _generate_streamlines(self):
             directions = dg.initial_direction(s)
             directions = directions[:max_cross]
             for first_step in directions:
-                stepsF = local_tracker(dg, tc, s, first_step, vs, F, ss, fixed)
-                if stepsF < 0:
+                stepsF, tissue_class = local_tracker(dg, tc, s, first_step,
+                                                     vs, F, ss, fixed)
+                # enum TissueClass (tissue_classifier.pyd) is not accessible
+                # from here. To be changed when minimal cyhton version > 0.21.
+                # cython 0.21 - cpdef enum to export values into Python-level namespace
+                # https://github.com/cython/cython/commit/50133b5a91eea348eddaaad22a606a7fa1c7c457
+                if not (self.return_all or
+                        tissue_class == 2 or  # ENDPOINT
+                        tissue_class == -1):  # OUTSIDEIMAGE
```

Comment:
I suspect that we'll eventually end up returning the endpoint codes so that user can include/exclude or otherwise post process the streamlines themselves.

I'm fine with leaving this as is and coming back to it if it's an issue for someone.